closure/goog/net/xhrio.js

```diff
@@ -332,12 +332,21 @@ goog.net.XhrIo.sendInstances_ = [];
  *     incomplete request will be aborted; 0 means no timeout is set.
  * @param {boolean=} opt_withCredentials Whether to send credentials with the
  *     request. Default to false. See {@link goog.net.XhrIo#setWithCredentials}.
+ * @param {goog.net.XmlHttpFactory=} opt_xmlHttpFactory Factory to use when
+ *     creating XMLHttpRequest objects
  * @return {!goog.net.XhrIo} The sent XhrIo.
  */
 goog.net.XhrIo.send = function(
     url, opt_callback, opt_method, opt_content, opt_headers,
-    opt_timeoutInterval, opt_withCredentials) {
-  var x = new goog.net.XhrIo();
+    opt_timeoutInterval, opt_withCredentials, opt_xmlHttpFactory) {
+
+  var x;
```

Comment:
I'd prefer the ternary operator instead of the `if` to avoid the uninitialized variable. See [5.1.3 Declared when needed, initialized as soon as possible](https://google.github.io/styleguide/jsguide.html#features-declared-when-needed) or [Deferred initialization](https://google.github.io/styleguide/javascriptguide.xml?showone=Deferred_initialization#Deferred_initialization).